src/transformers/models/bridgetower/modeling_bridgetower.py

```diff
@@ -0,0 +1,1006 @@
+        )[0]
+        cross_image_feats = self.cross_modal_image_layers[0](
+            cross_modal_image,
+            cross_modal_text,
+            attention_mask=extend_image_masks,
+            encoder_attention_mask=extend_text_masks,
+        )[0]
+
+        link_layer_index = 0
+
+        # link tower fusion
+        for i in range(split_index, len(self.text_transformer.encoder.layer)):
+            text_embeds = self.text_transformer.encoder.layer[i](text_embeds, extend_text_masks)[0]
+            image_embeds = self.vit_model.visual.transformer.resblocks[i](image_embeds).type(self.vit_model.dtype)
+            image_embeds_with_ln = (
+                self.cross_modal_image_transform(self.vit_model.visual.forward_post(image_embeds))
+                + image_token_type_embeddings
+            )
+
+            text_link_tower = self.cross_modal_text_link_tower[link_layer_index]
+            image_link_tower = self.cross_modal_image_link_tower[link_layer_index]
+
+            cross_text_feats_ = text_link_tower(
+                self.cross_modal_text_transform(text_embeds) + text_token_type_embeddings,
+                cross_text_feats,
+                extend_text_masks,
+            )
+            if irtr_len > 0:
+                cross_image_feats_ = image_link_tower(
+                    image_embeds_with_ln.unsqueeze(1).expand(_bs, irtr_len, _L, _D).reshape(-1, _L, _D),
+                    cross_image_feats,
+                    extend_image_masks,
+                )
+            else:
+                cross_image_feats_ = image_link_tower(image_embeds_with_ln, cross_image_feats, extend_image_masks)
+
+            cross_text_feats = self.cross_modal_text_layers[link_layer_index + 1](
+                cross_text_feats_,
+                cross_image_feats_,
+                attention_mask=extend_text_masks,
+                encoder_attention_mask=extend_image_masks,
+            )[0]
+            cross_image_feats = self.cross_modal_image_layers[link_layer_index + 1](
+                cross_image_feats_,
+                cross_text_feats_,
+                attention_mask=extend_image_masks,
+                encoder_attention_mask=extend_text_masks,
+            )[0]
+
+            link_layer_index += 1
+
+        text_feats, image_feats = cross_text_feats, cross_image_feats
+        cls_feats = self.get_cls_feats(text_feats, image_feats)
+
+        if not return_dict:
+            return tuple(v for v in [text_feats, image_feats, cls_feats] if v is not None)
+
+        return BridgeTowerModelOutput(
+            text_feats=text_feats,
+            image_feats=image_feats,
+            pooler_output=cls_feats,
+        )
+
+    def get_cls_feats(self, text_feats, image_feats):
+        cls_feats_text = self.cross_modal_text_pooler(text_feats)
+        cls_feats_image = self.cross_modal_image_pooler(image_feats)
+        return torch.cat([cls_feats_text, cls_feats_image], dim=-1)
+
+    def freeze_layers(
```

Comment:
Is this function used?

Reply:
This is an optional function that can be used during training by users. 
eg. 
```
model = BridgeTowerModel.from_pretrained('...')
model.train()
model.freeze_layers()
```

Reply:
I see, thanks a lot for the clarification!